src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsCodeFixProvider.cs

```diff
@@ -0,0 +1,108 @@
+        public CSharpUsePatternCombinatorsCodeFixProvider()
+            : base(supportsFixAll: true)
+        {
+        }
+
+        private static SyntaxKind MapToSyntaxKind(BinaryOperatorKind kind)
+            => kind switch
+            {
+                BinaryOperatorKind.LessThan => SyntaxKind.LessThanToken,
+                BinaryOperatorKind.GreaterThan => SyntaxKind.GreaterThanToken,
+                BinaryOperatorKind.LessThanOrEqual => SyntaxKind.LessThanEqualsToken,
+                BinaryOperatorKind.GreaterThanOrEqual => SyntaxKind.GreaterThanEqualsToken,
+                _ => throw ExceptionUtilities.UnexpectedValue(kind)
+            };
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.UsePatternCombinatorsDiagnosticId);
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.RequireSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var location = diagnostic.Location;
+                var expression = editor.OriginalRoot.FindNode(location.SourceSpan, getInnermostNodeForTie: true);
+                var operation = semanticModel.GetOperation(expression, cancellationToken);
+                RoslynDebug.AssertNotNull(operation);
+                var pattern = CSharpUsePatternCombinatorsAnalyzer.Analyze(operation);
+                RoslynDebug.AssertNotNull(pattern);
+                var patternSyntax = AsPatternSyntax(pattern).WithAdditionalAnnotations(Formatter.Annotation);
+                editor.ReplaceNode(expression, IsPatternExpression((ExpressionSyntax)pattern.Target.Syntax, patternSyntax));
+            }
+        }
+
+        private static PatternSyntax AsPatternSyntax(AnalyzedPattern pattern)
+            => pattern switch
+            {
+                Binary p => BinaryPattern(
+                    p.IsDisjunctive ? SyntaxKind.OrPattern : SyntaxKind.AndPattern,
+                    AsPatternSyntax(p.Left).Parenthesize(),
+                    Token(p.Token.LeadingTrivia, p.IsDisjunctive ? SyntaxKind.OrKeyword : SyntaxKind.AndKeyword,
+                        TriviaList(p.Token.GetAllTrailingTrivia())),
+                    AsPatternSyntax(p.Right).Parenthesize()),
+                Constant p => ConstantPattern(p.ExpressionSyntax),
```

Comment:
i wonder if this needs tobe parenthesized.  for example, if the expression was some constant like `A|B`.